WalletWasabi.Gui/Converters/RunOnSystemStartupAfterStartJsonConverter.cs

```diff
@@ -0,0 +1,28 @@
+using System;
+using System.Threading;
+using Newtonsoft.Json;
+using WalletWasabi.Helpers;
+
+namespace WalletWasabi.Gui.Converters
+{
+	public class RunOnSystemStartupAfterStartJsonConverter : JsonConverter
+	{
+		public override bool CanConvert(Type objectType)
+		{
+			return objectType == typeof(bool);
+		}
+
+		public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+		{
+			using var cts = new CancellationTokenSource(10000);
```

Comment:
Converter should just convert and not read from the registry. Also keep in mind - in case of any error, still the show must go on. I prefer the previous implementation - this is unexpected.

Reply:
I did not check what `GetCurrentValueAsync` does, I just sketched a sample of a possible solution. 
As far as I understand the problem is after deserialization the value of `RunOnSystemStartup` might be not correct.
Basically it is the same situation as at the `WindowState` in the UiConfig, where we don't allow the config to be created with a minimized window state.

So it looked like a good possible solution because creating a UiConfig then modify some of its value will cause problems.
All the configs are expecting to have correct values after deserialization. 

See this example:
```
string dataDir = EnvironmentHelpers.GetDataDir(Path.Combine("WalletWasabi", "Client"));
var (uiConfig, config) = LoadOrCreateConfigs(dataDir);

uiConfig.RunOnSystemStartup = true; // This would be the correction

var isDifferent = uiConfig.CheckFileChange(); // Will return true, but it should be false.
```

I agree with you on reading from a registry is too heavy for a converter, this is my fault, I should have dived deeper before I suggest enything.